### PR TITLE
27 - Login/register with API from the server

### LINKED ISSUE #27: Login/Register page
- Login with username & password
- Register new user account

### PR DESCRIPTION
**Pull request by tientrinh21**
User would be able to:
- Login: username + password
- Register: username + password + nationality (optional) + home university (optional)

The notice regarding the operation is successful or failed will be displayed on the right bottom corner of the screen.

### CODE REVIEW

File: server/.DS_Store
Comment (wumbus) [inline server/.DS_Store:1]: Should this be here? Git ignore?

File: client/app/(auth)/sign-in/page.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+export default function SignInPage() {
+  const router = useRouter()
+  const setIsAuth = useSetAtom(authAtom)
+
+  // Define form
+  const form = useForm<LoginFormSchema>({
+    resolver: zodResolver(loginFormSchema),
+    defaultValues: {
+      username: 'test',
```
Comment (wumbus) [inline client/app/(auth)/sign-in/page.tsx]: Why is it prefilled?
image

Comment (wumbus) [reply client/app/(auth)/sign-in/page.tsx]: image

**Review by wumbus — CHANGES_REQUESTED** [review]
Some questions:
- If I am logged in and then press my page / profile I get logged out, possibly because the page does not exists. Is this intended to happen?
- What is the difference between my page and profile? Why two buttons?

The rest is nicely done. Good work!

File: server/.DS_Store
Comment (tientrinh21) [reply server/.DS_Store:1]: I am not quite sure about the file, let me add the to gitignore

File: client/app/(auth)/sign-in/page.tsx (hunk 1 above)
Comment (tientrinh21) [reply client/app/(auth)/sign-in/page.tsx]: Oh I prefilled it for testing, shall I get rid of it?

**Comment by tientrinh21** [comment]
> Some questions:
> 
> * If I am logged in and then press my page / profile I get logged out, possibly because the page does not exists. Is this intended to happen?
> * What is the difference between my page and profile? Why two buttons?
> 
> The rest is nicely done. Good work!

- The problem that auto logged out was my bad, I create an `eventListener` on storage to check if user is logged in, but it doesn't change on page reload, so that's why the Sign in button appears. I fixed it with `useEffect` on the `auth.ts` file.
- If I remembered correctly, `My Page` is use for pinning the universities that you are considering for your exchange, my profile is just a page where you update your personal information (e.g. password, nationality, etc.)

**Review by wumbus — APPROVED** [review]

**Pull request merged**